src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -9676,3 +9676,48 @@ void MacroAssembler::check_stack_alignment(Register sp, const char* msg, unsigne
   stop(msg);
   bind(L_stack_ok);
 }
+
+void MacroAssembler::fast_lock_impl(Register obj, Register hdr, Register thread, Register tmp, Label& slow) {
+  assert(hdr == rax, "header must be in rax for cmpxchg");
+  assert_different_registers(obj, hdr, thread, tmp);
+
+  // First we need to check if the lock-stack has room for pushing the object reference.
+  cmpl(Address(thread, JavaThread::lock_stack_offset_offset()), LockStack::end_offset());
+  jcc(Assembler::greaterEqual, slow);
+
+  // Now we attempt to take the fast-lock.
+  // Clear lowest two header bits (locked state).
+  andptr(hdr, ~(int32_t)markWord::lock_mask_in_place);
+  movptr(tmp, hdr);
+  // Set lowest bit (unlocked state).
+  orptr(hdr, markWord::unlocked_value);
+  lock();
+  cmpxchgptr(tmp, Address(obj, oopDesc::mark_offset_in_bytes()));
+  jcc(Assembler::notEqual, slow);
+
+  // If successful, push object to lock-stack.
+  movl(tmp, Address(thread, JavaThread::lock_stack_offset_offset()));
+  movptr(Address(thread, tmp, Address::times_1), obj);
```

Comment:
Minor: I think `Address::times_1` is unnecessary in cases like this. (You might want to grep for other instances of this).